### ISSUE TITLE
CORS XHR Support

### ISSUE THREAD

**Issue by amiri-ge** [cloudfoundry/uaa#173]
Modern browsers include a "X-Requested-With: XMLHttpRequest" header when RESTful calls are performed through the XMLHttpRequest library. Since the header is not allowed by default CORS policies, requests that would otherwise be allowed by CORS standard policy will fail across different origins, thus providing a mitigation against CSRF.

However, when using implicit grant, it is useful to allow cross-origin XHR requests for endpoints such as /logout.do and /userinfo. At the same time, one should not allow this across the board. We've developed a CORS filter that allows these requests based on a whitelist of both allowed endpoints and origins but are hesitant to submit this as a pull request because:

1. We believe it might be something you are already be working on.
2. Given that CORS is enforced at the browser, it's a difficult feature to test.

In summary, if you have plans to support such a feature please let us know. If not we would like to request this as a future feature. Finally, let us know if you would be interested in a pull request for this feature.

**Comment by fhanik** [comment]
1. We are not working on such feature
2. Take a look at our integration tests for how we test browser and UI features. It may be possible to leverage Selenium to achieve your goals

It would be easier for us to see the PR to understand what you are trying to achieve.

**Comment by amiri-ge** [comment]
@fhanik I submitted a pull request to provide more context for this issue.

**Comment by fhanik** [comment]
https://github.com/cloudfoundry/uaa/commit/6b5cd4867a785f5e8a27621c320d77431f75a337
